tests/data/cli/null-defaults.proto

```diff
@@ -0,0 +1,11 @@
+syntax = "proto2";
```

Comment:
would it be only for `proto2`?
does it make sense to test it against `proto3` too?

Reply:
`--null-defaults` is already the default behavior for proto3 according to @alexander-fenster  : 

> For proto3 optional, I think I got it right in #1584 and #1597 (always assigning them to null).

Reply:
Actually it's not exactly how it works. Internally missing values might be represented by undefined or null, but the getters pervert his behavior, returning non-null defaults.

Reply:
Thanks for spotting this @castarco. We mostly use static objects (generated by `pbjs -t static-module`) and the default values are set to `null` there for optional fields; I haven't seen any bad consequences of the `field.js` code but I guess we just don't have this use case in our code base. I'll look into this, I believe it should not be hard to fix it.

Reply:
This is what I was talking about (`proto3` optional fields are internally `oneof` members so get `null` as their default values):

https://github.com/protobufjs/protobuf.js/blob/master/cli/targets/static.js#L414

Reply:
I'm maintaining a code generator ( https://github.com/join-com/protoc-gen-ts/ , quite opinionated, for the specific uses of my company, although it would be usable for others too ), and I had to implement a fix today because of this:
https://github.com/join-com/protoc-gen-ts/pull/37

Reply:
_today I learned: you need to link to the specific commit SHA to make GitHub magic work :)_

Reply:
Is this how it is supposed to work? To always use `oneOf` fields if we want truly nullable fields in proto3? 😢

Reply:
Yes, this is how this is supposed to work in proto3. Original design: https://github.com/protocolbuffers/protobuf/blob/master/docs/implementing_proto3_presence.md - I just implemented it as is.

The thing is, message fields are always optional. The main reason of having optional primitive fields is to distinguish between integer 0 and "unset" value, same for strings and booleans. Putting them into a hidden `oneof` is apparently the easiest non-breaking solution.

Reply:
`optional` fields are supported in proto3 starting from `protoc` 3.12 (under the flag) and IIRC 3.15 without the flag.